lib/date.js

```diff
@@ -98,8 +98,17 @@ module.exports = (key, options) => {
         label: label === labelKey ? v.label : label
       });
     }));
+    const conditionalTranslate = (fieldKey) => {
+      let result = req.translate(fieldKey);
+      if (result === fieldKey) {
+        result = null;
+      }
+      return result;
+    };
```

Comment:
exactly that! Pure functions enforce much simpler code, as you can see exactly what the function has access to. Helpers like this perform one action and don't have any side effects.

```js
getValues(req, res, next) {
  const getVal() {
    return req.sessionModel.get(req.params.value);
  }
  getVal()
}

// vs
...
const getVal(val, model) {
  return model.get(val);
}

class blah {
  getValues(req, res, next) {
    const val = getVal(req.params.value, req.sessionModel);
  }
}
```